### PR TITLE
Requests console no longer spam "New materials available"

### LINKED ISSUE #19124: Requests console are spamming "new minerals available"
Even if there's no mineral available. Only goes for requests console that gives the message when new minerals are available.

### PR DESCRIPTION
**Pull request by lzimann**
Fixes #19124

### CODE REVIEW

File: code/modules/mining/machine_redemption.dm [hunk 1]
```diff
@@ -59,6 +59,11 @@
 			var/obj/item/stack/sheet/s = new processed_sheet(src,0)
 			s.amount = 0
 			stack_list[processed_sheet] = s
+			if(s.name != "glass" && s.name != "metal") // no need to announce metal/glass
+				var/msg = "[capitalize(s.name)] sheets now available at Cargo Bay."
+				for(var/obj/machinery/requests_console/D in allConsoles)
+					if(D.department == "Science" || D.department == "Robotics" || D.department == "Research Director's Desk" || D.department == "Chemistry" || D.department == "Bar" )
```
Comment (GunHog) [inline code/modules/mining/machine_redemption.dm]: Geez, this could really use a var or something you only have to check once per console.

**Comment by Shapsy** [comment]
Please speedmerge or something holy shit this is awful

File: code/modules/mining/machine_redemption.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/mining/machine_redemption.dm]: i thought the same thing but got lazy

**Comment by optimumtact** [comment]
This shouldn't even live in process_sheet, it should just collect a count of all sheet types and send them at the end of the process.

File: code/modules/mining/machine_redemption.dm [hunk 2]
```diff
@@ -88,13 +93,6 @@
 					else
 						process_sheet(O)
 						i++
-		var/msg = "Now available in the Cargo Bay: \n"
```
Comment (optimumtact) [inline code/modules/mining/machine_redemption.dm]: Just add
    
    if(i > 0)

that way the message is only sent if there were any new ores processed this tick, which is probalby what was intended.

**Pull request merged**